src/main/java/org/wildfly/security/auth/realm/ldap/SimpleDirContextFactoryBuilder.java

```diff
@@ -240,6 +257,7 @@ private DirContext createDirContext(String securityPrincipal, char[] securityCre
             env.put(InitialDirContext.SECURITY_PRINCIPAL, securityPrincipal);
             env.put(InitialDirContext.SECURITY_CREDENTIALS, String.valueOf(securityCredential));
             env.put(InitialDirContext.REFERRAL, mode == null ? ReferralMode.IGNORE.getValue() : mode.getValue());
+            if (socketFactory != null) env.put(SOCKET_FACTORY, ElytronSSLSocketFactory.class.getName());
```

Comment:
I guess for this to work, our socket factory has to be visible from the TCCL, correct?